bigtable/client/internal/table.h

```diff
@@ -94,19 +98,27 @@ class Table {
 
   template <typename RPCRetryPolicy, typename RPCBackoffPolicy,
             typename IdempotentMutationPolicy>
-  Table(std::shared_ptr<DataClient> client, std::string app_profile_id,
-        std::string const& table_id, RPCRetryPolicy retry_policy,
-        RPCBackoffPolicy backoff_policy,
+  Table(std::shared_ptr<DataClient> client,
+        bigtable::AppProfileId app_profile_id, std::string const& table_id,
+        RPCRetryPolicy retry_policy, RPCBackoffPolicy backoff_policy,
         IdempotentMutationPolicy idempotent_mutation_policy)
       : client_(std::move(client)),
         app_profile_id_(std::move(app_profile_id)),
-        table_name_(TableName(client_, table_id)),
+        table_name_(bigtable::TableId(TableName(client_, table_id))),
         rpc_retry_policy_(retry_policy.clone()),
         rpc_backoff_policy_(backoff_policy.clone()),
         metadata_update_policy_(table_name(), MetadataParamTypes::TABLE_NAME),
         idempotent_mutation_policy_(idempotent_mutation_policy.clone()) {}
 
-  std::string const& table_name() const { return table_name_; }
+  std::string const& table_name() const { return table_name_.get(); }
+
+  template <typename Request>
```

Comment:
This should not be in the public section.  We do not expect or want application developers to call it, right?

Reply:
Maybe it should be a standalone function in the `internal` namespace?

Reply:
Done.